File: 0_Project_CURRENT/Assets/scripts/GUIdisable.js
var guiQuadObjScript : GUIQuadObj;
var shipObj : GameObject;
var ship : MoveShip;
var guiObj : GameObject;
var gui : GUIScript;

function Awake() { 
	guiQuadObjScript = this.GetComponent("GUIQuadObj");
	shipObj = GameObject.Find("Ship");
	ship = shipObj.GetComponent("MoveShip");
	guiObj = GameObject.Find("GUI_1");
	gui = guiObj.GetComponent("GUIScript");
}


function Update () {
	if(gui.state=="play" && !ship.state.cruising){
		guiQuadObjScript.Enabled=true;
		guiQuadObjScript.Visible=true;	
	}else{
		guiQuadObjScript.Enabled=false;
		guiQuadObjScript.Visible=false;	
	}	
}

File: 0_Project_CURRENT/Assets/scripts/GUIdisable.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class GUIdisable : MonoBehaviour
{

	GUIQuadObj guiQuadObjScript;
	GameObject shipObj;
	MoveShip ship;
	GameObject guiObj;
	GUIScript gui;

	void Awake ()
	{ 
		guiQuadObjScript = this.GetComponent<GUIQuadObj> ();
		shipObj = GameObject.Find ("Ship");
		ship = shipObj.GetComponent<MoveShip> ();
		guiObj = GameObject.Find ("GUI_1");
		gui = guiObj.GetComponent<GUIScript> ();
	}


	void Update ()
	{
		if (gui.state == "play" && ship.state == MoveShip.State.Normal) {
			guiQuadObjScript.Enabled = true;
			guiQuadObjScript.Visible = true;	
		} else {
			guiQuadObjScript.Enabled = false;
			guiQuadObjScript.Visible = false;	
		}	
	}

}
